saylua/modules/messages/models/db.py

```diff
@@ -21,12 +20,13 @@ class ConversationUser(db.Model):
 
     title = db.Column(db.String(256))
     unread = db.Column(db.Boolean, default=False)
+    deleted = db.Column(db.Boolean, default=False)
     last_updated = db.Column(db.DateTime(timezone=True), server_default=db.func.now())
 
     def url(self):
-        if not self.is_read:
-            return '/conversation_read/' + self.conversation_id + '/'
-        return '/conversation/' + self.conversation_id + '/'
+        if self.unread:
+            return '/conversation_read/' + str(self.conversation_id) + '/'
```

Comment:
I was the one who wrote the original code here. It was meant to be an optimization because the conversation_read handle looks up the UserConversation object and marks it as read if it's not read. I think this applied more to ndb than now though. 

In the old ndb schema, I had denormalized things so that when you were viewing the actual contents of the conversation, there was no need to look up the UserConversation object. But now you have to look it up anyways to verify that the user has access to the conversation they're viewing, so this isn't really an optimization at all anymore.